### ISSUE TITLE
Optimization, Loss Functions and Regularization

### ISSUE THREAD

**Issue by Rishikesh75** [Rishikesh75/backend-devops-lab#18]
## Parent Epic
Part of #9 — Deep Learning Interview Preparation.

## Topics
- Cross-entropy and MSE
- SGD, Momentum, Adam
- Learning rate schedules
- Batch size
- Dropout
- Weight decay
- Batch normalization
- Early stopping

## Interview Focus
Compare optimizers and explain how loss functions, learning rate and regularization affect training.
